src/hooks/useAuth.ts

```diff
@@ -51,7 +54,12 @@ const useAuth = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  return { login, logout: deactivate }
+  const logout = useCallback(() => {
+    dispatch(profileClear())
+    deactivate()
+  }, [deactivate, dispatch])
+
+  return { login, logout }
```

Comment:
Why the need for `useCallback`?

Reply:
So that we don't create a new reference every time this hook is invoked and accidentally trigger unnecessary recalculations / re-renders somewhere.

If that's not a concern I could of course remove it.

Reply:
I would rather disable the eslint rule and put en empty array to make sure the fonction is never re created

Reply:
>If that's not a concern I could of course remove it.

A valid concern but is it happening? Both of the those functions should keep their reference.

>I would rather disable the eslint rule and put en empty array to make sure the fonction is never re created

While this case might be okay I think it will lead to bugs. As stated on the React docs it is recommended to include all dependencies.

Reply:
> A valid concern but is it happening? Both of the those functions should keep their reference.

@hachiojidev both of these functions do keep their reference, don't they?
Could you please elaborate more on the desired outcome?

> I would rather disable the eslint rule and put en empty array to make sure the fonction is never re created

@RabbitDoge but we do want this function re-created if the dependencies change, otherwise if `dispatch` or `deactivate` change their reference, but we still call them by their old ones, then our function might not work properly anymore.